src/api/__tests__/hooks.unit.test.ts.tsx

```diff
@@ -1,533 +1,315 @@
-      });
-    });
+    expect(mockAxios.lastReqGet()).toBeUndefined();
   });
 
-  it("returns an error on an unsuccessful response", async () => {
-    const { result } = renderHook(() => h.useApiPost("test"), { wrapper: AuthProvider });
-    act(() => {
-      mockAxios.mockResponse({
-        data: { success: true, data: testUser }
-      });
-    });
+  it("executes the request when the callback is called", async () => {
+    const get = retrieveFactory<string>("bingo/");
 
-    const [post] = result.current;
-    let dataPromise: any;
+    const { result } = renderHook(() => useAPILazy(get));
 
-    act(() => {
-      dataPromise = post("test data");
-    });
-
-    act(() => {
-      mockAxios.mockError(makeTestError(500, ERRORS.__TESTING));
-    });
+    let [exec, info] = result.current;
 
     await act(async () => {
-      try {
-        await dataPromise;
-        expect("Error was not thrown by useApiPost").toBe(false);
-      } catch (e) {
-        expect(e).toBe(ERRORS.__TESTING);
-      }
+      const resp = exec({ id: "15" });
+      expect(mockAxios.lastReqGet().url).toEqual("bingo/15/");
+
+      mockAxios.mockResponseFor(
+        {
+          method: "GET",
+          url: "bingo/15/",
+        },
+        {
+          data: MOCK_RESPONSE,
+        }
+      );
+      expect(await resp).toEqual(MOCK_RESPONSE);
     });
   });
 
-  it("returns a generic error message on an unsuccessful response without a specific error message", async () => {
-    const { result } = renderHook(() => h.useApiPost("test"), { wrapper: AuthProvider });
-    act(() => {
-      mockAxios.mockResponse({
-        data: { success: true, data: testUser }
-      });
-    });
+  it("correctly updates RequestState", async () => {
+    const get = retrieveFactory<string>("bingo/");
 
-    const [post] = result.current;
-    let dataPromise: any;
+    const { result } = renderHook(() => useAPILazy(get));
 
-    act(() => {
-      dataPromise = post("test data");
-    });
+    let [exec, info] = result.current;
+    expect(info.state).toBe(RequestState.NotStarted);
 
     act(() => {
-      mockAxios.mockError(makeTestError(500, null));
+      exec({ id: "15" });
     });
 
+    [exec, info] = result.current;
+    expect(info.state).toBe(RequestState.Running);
+
     await act(async () => {
-      try {
-        await dataPromise;
-        expect("Error was not thrown by useApiPost").toBe(false);
-      } catch (e) {
-        expect(e).toEqual({
-          status_code: 500,
-          detail: [ERRORS.REQUEST.DID_NOT_SUCCEED]
-        });
-      }
+      mockAxios.mockResponseFor(
+        {
+          method: "GET",
+          url: "bingo/15/",
+        },
+        {
+          data: MOCK_RESPONSE,
+        }
+      );
     });
+
+    [exec, info] = result.current;
+    expect(info.state).toBe(RequestState.Complete);
   });
 });
 
-describe("useApiPatch", () => {
-  afterEach(() => {
-    mockAxios.reset();
+describe("useAPILazyCSRF", () => {
```

Comment:
Missing test cases for unsuccessful requests

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/33/commits/9522bc49a1895c48b860f58aa1c765cbf918e6ad